app/models/deck.rb

```diff
@@ -0,0 +1,10 @@
+class Deck < ActiveRecord::Base
+  has_many :cards, dependent: :destroy
+  belongs_to :user
+
+  validates_uniqueness_of :current, conditions: -> { where(current: true) }
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.